ui/pages/Applications.tsx

```diff
@@ -16,24 +18,47 @@ type Props = {
 
 function Applications({ className }: Props) {
   const [applications, setApplications] = React.useState<Application[]>([]);
-  const { listApplications, loading } = useApplications();
+  const { applicationsClient, doAsyncError } = React.useContext(AppContext);
+  const [loading, setLoading] = React.useState(false);
+
+  const getApps = () => {
+    setLoading(true);
+    applicationsClient
+      .ListApplications({ namespace: "wego-system" })
+      .then((res) => setApplications(res.applications))
+      //how deep are we going with these error messages? Should I have one for 500 and one for 400?
+      .catch((err) =>
+        doAsyncError(
+          "Unable to retrieve applications",
+          "Something went wrong - please try again later"
```

Comment:
We want to give the user some breadcrumbs to follow:

```suggestion
          err.message
```

Reply:
So I can use err.message and err.detail? Do I have to check if it's a 500 first and handle that case (if there's no message?)

Reply:
A 500 will still have a message. Every error will probably have a message.

Reply:
Awesome I'll change it 🌞